gammapy/spectrum/models.py

```diff
@@ -574,30 +577,129 @@ class TableModel(SpectralModel):
         Array with the values of the model at energies ``energy``.
     amplitude : float
         Model amplitude that is multiplied to the supplied arrays. Defaults to 1.
+    intery : boolean
+        interpolation can be done linearly or in logarithm
     """
 
-    def __init__(self, energy, values, amplitude=1):
+    def __init__(self, energy, values, amplitude=1, scale_logy=True):
         from scipy.interpolate import interp1d
         self.parameters = Bunch(amplitude=amplitude)
         self.energy = energy
         self.values = values
-
+        self.scale_logy = scale_logy
+        
         loge = np.log10(self.energy.to('eV').value)
         try:
             self.unit = self.values.unit
-            logy = np.log10(self.values.value)
+            if scale_logy is True:
+                y = np.log10(self.values.value)
+            else:
+                y = self.values.value
         except AttributeError:
             self.unit = u.Unit('')
-            logy = np.log10(self.values)
-
-        self.interplogy = interp1d(loge,
-                                   logy,
+            if scale_logy is True:
+                y = np.log10(self.values)
+            else:
+                y = self.values
+        self.interpy = interp1d(loge,
+                                   y,
                                    fill_value=-np.Inf,
                                    bounds_error=False,
                                    kind='cubic')
 
+    @classmethod
+    def from_file(cls, filename, param):
```

Comment:
What is `param`???

Mention it with one line (or add a `Parameters` section) in the docstring?